markets/perps-market/contracts/modules/PerpsMarketFactoryModule.sol

```diff
@@ -101,11 +108,37 @@ contract PerpsMarketFactoryModule is IPerpsMarketFactoryModule {
 
     function name(uint128 perpsMarketId) external view override returns (string memory) {
         // todo: set name on initialize?
+        perpsMarketId; // silence unused variable warning
         return "Perps Market";
     }
 
     function reportedDebt(uint128 perpsMarketId) external view override returns (uint256) {
-        // TODO
+        PerpsMarketFactory.Data storage factory = PerpsMarketFactory.load();
+
+        if (factory.perpsMarketId == perpsMarketId) {
+            // debt is the total debt of all markets
+            // can be computed as total collateral value - sum_each_market( debt )
+            uint totalCollateralValue = GlobalPerpsMarket.load().totalCollateralValue();
+            int totalDebt;
+
+            SetUtil.UintSet storage activeMarkets = GlobalPerpsMarket.load().activeMarkets;
+            uint256 activeMarketsLength = activeMarkets.length();
+            for (uint i = 1; i <= activeMarketsLength; i++) {
+                uint128 marketId = activeMarkets.valueAt(i).to128();
+                totalDebt += PerpsMarket.load(marketId).marketDebt(
+                    PerpsPrice.getCurrentPrice(marketId)
+                );
+            }
+
+            int reportedDebt = totalCollateralValue.toInt() + totalDebt;
+            if (reportedDebt < 0) {
+                return 0;
+            } else {
+                return reportedDebt.toUint();
+            }
+        }
+
+        // TODO Should revert if perpsMarketId is not correct???
```

Comment:
0 is fine.  we probably should never revert on reported debt since it could brick the core system